consensus/hotstuff/safetyrules/safety_rules.go

```diff
@@ -45,42 +46,53 @@ func New(
 //  * (nil, model.NoVoteError): If the voter decides that it does not want to vote for the given block.
 //    This is a sentinel error and _expected_ during normal operation.
 // All other errors are unexpected and potential symptoms of uncovered edge cases or corrupted internal state (fatal).
```

Comment:
We should take a pass over the godoc here, eg `SafetyRules will ask Forks whether a vote is a safe node or not` is outdated.